src/_pytest/hookspec.py

```diff
@@ -260,7 +273,9 @@ def pytest_pycollect_makemodule(path, parent):
 
 
 @hookspec(firstresult=True)
-def pytest_pycollect_makeitem(collector, name, obj):
+def pytest_pycollect_makeitem(
+    collector: "PyCollector", name: str, obj
+) -> "Union[None, Item, Collector, List[Union[Item, Collector]]]":
     """ return custom item/collector for a python object in a module, or None.
 
     Stops at first non-None result, see :ref:`firstresult` """
```

Comment:
Despite what the docs say I think this actually returns only Items, not Collectors, doesn't it?

Reply:
Oh, it might be `Class(PyCollector)` for example..  (which however only returns a single `Instance` via `collect()` then, but that's for something else).  So this is likely good then.